3-Network-Configuration/3.5-Firewall-Configuration.bats

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env bats
+
+load IPv6-helper
+
+# 3.5.1 Configure UncomplicatedFirewall
+
+@test "3.5.1.1 Ensure ufw is installed (Automated)" {
+    [[ $(dpkg -s ufw | grep 'Status: install') == "Status: install ok installed" ]]
+}
+
+@test "3.5.1.2 Ensure iptables-persistent is not installed with ufw (Automated)" {
+    run dpkg-query -s iptables-persistent
+    [ $status -eq 1 ]
+    [[ "$output" == *"package 'iptables-persistent' is not installed and no information is available"* ]]
+}
+
+@test "3.5.1.3 Ensure ufw service is enabled (Automated)" {
+    [[ $(systemctl is-enabled ufw) == "enabled" ]]
+    [[ $(ufw status | grep Status) == "Status: active" ]]
+}
+
+@test "3.5.1.4 Ensure ufw loopback traffic is configured (Automated)" {
+    run bash -c "ufw status verbose"
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"Anywhere on lo"*"ALLOW IN"*"Anywhere"* ]]
+    [[ "$output" == *"Anywhere"*"DENY IN"*"127.0.0.0/8"* ]]
+    [[ "$output" == *"Anywhere (v6) on lo"*"ALLOW IN"*"Anywhere (v6)"* ]]
+    [[ "$output" == *"Anywhere (v6)"*"DENY IN"*"::1"* ]]
+    [[ "$output" == *"Anywhere"*"ALLOW OUT"*"Anywhere on lo"* ]]
+    [[ "$output" == *"Anywhere (v6)"*"ALLOW OUT"*"Anywhere (v6) on lo"* ]]
+}
+
+@test "3.5.1.5 Ensure ufw outbound connections are configured (Manual)" {
+    skip "This audit has to be done manually"
+}
+
+@test "3.5.1.6 Ensure ufw firewall rules exist for all open ports (Manual)" {
+    skip "This audit has to be done manually"
+}
+
+@test "3.5.1.7 Ensure ufw default deny firewall policy (Automated)" {
+    run bash -c "ufw status verbose | grep -i default"
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"deny (incoming)"* || "$output" == *"reject (incoming)"* ]]
+    [[ "$output" == *"deny (outgoing)"* || "$output" == *"reject (outgoing)"* ]]
+    [[ "$output" == *"deny (routed)"* \
+     || "$output" == *"reject (routed)"* \
+     || "$output" == *"disabled (routed)"* ]]
+}
+
+# 3.5.2 Configure nftables
+
+@test "3.5.2.1 Ensure nftables is installed (Automated)" {
+    run bash -c "dpkg -s nftables | grep 'Status: install'"
+    [[ "$output" == "Status: install ok installed" ]]
+}
+
+@test "3.5.2.2 Ensure ufw is uninstalled or disabled with nftables (Automated)" {
+    run bash -c "dpkg -s ufw | grep 'dpkg-query'"
```

Comment:
This does not match the command from the CIS document (`dpkg-query -s ufw | grep 'Status: install ok installed'`)